src/stages/normalize/patchers/FunctionPatcher.js

```diff
@@ -0,0 +1,52 @@
+import NodePatcher from './../../../patchers/NodePatcher.js';
+import type { Node, ParseContext, Editor } from './../../../patchers/types.js';
+
+export default class FunctionPatcher extends NodePatcher {
+  parameters: Array<NodePatcher>;
+  body: ?NodePatcher;
+  
+  constructor(node: Node, context: ParseContext, editor: Editor, parameters: Array<NodePatcher>, body: ?NodePatcher) {
+    super(node, context, editor);
+    this.parameters = parameters;
+    this.body = body;
+  }
+
+  patchAsExpression() {
+    // To avoid knowledge of all the details how assignments can be nested in nodes,
+    // we add a callback to the function node before patching the parameters and remove it afterwards.
+    // This is detected and used by the MemberAccessOpPatcher to claim a free binding for this parameter
+    // (from the functions scope, not the body's scope)
+
+    let assignments = [];
+    this.node._assignMember = function(memberName: string){
```

Comment:
this *is* ugly, might work nicer, if patch is a generator function, so there could be easier communication between caller and nested callee…